packages/terra-form/src/Control.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import 'terra-base/lib/baseStyles';
+
+import Input from './Input';
+
+import './Control.scss';
+
+const propTypes = {
+  /**
+   * The input type
+   */
+  type: PropTypes.oneOf(['checkbox', 'radio']).isRequired,
+  /**
+   * The checked property of the Input element. Use this to generate a controlled Control Element
+   */
+  checked: PropTypes.bool,
+  /**
+   * The checked property of the Input element. Use this to generate an uncontrolled Control Element
+   */
+  defaultChecked: PropTypes.bool,
+  /**
+   * id of the input field
+   */
+  id: PropTypes.string,
+  /**
+   * Additional attributes for the input object
+   */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the form element is inline or not
+   */
+  isInline: PropTypes.bool,
+  /**
+   * Text of the label
+   */
+  labelText: PropTypes.node,
+  /**
+   * Additional objects for the text object
+   */
+  labelTextAttrs: PropTypes.object,
+  /**
+   * Name of the input attribute
+   */
+  name: PropTypes.string,
+  /**
+   * The Value of the input element
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  type: null,
+  checked: undefined,
+  defaultChecked: undefined,
+  id: undefined,
+  inputAttrs: {},
+  isInline: false,
+  labelText: null,
+  labelTextAttrs: {},
+  name: null,
+  value: undefined,
+};
+
+const Control = ({ type, checked, defaultChecked, inputAttrs, id, isInline, labelText, labelTextAttrs, name, value, ...customProps }) => {
+  const labelClassNames = classNames(
+    'terra-Form-control',
+    { 'terra-Form-control--inline': isInline },
+    customProps.className,
+  );
+
+  const labelTextClasses = classNames(
+    'terra-Form-control-label-text',
+    labelTextAttrs.className,
+  );
+
+  const controlInputAttrs = Object.assign({}, inputAttrs);
+
+  if (checked !== undefined) {
+    controlInputAttrs.checked = checked;
+  } else {
```

Comment:
We talked in person. The default behavior for a react component when it receives undefined is to act as an uncontrolled value, and I'd like for consumers to be able to use the default react behavior for these inputs. As such, I believe it's best to keep it as undefined for the defaultChecked.